gpu_tests/test_checkpoint_saving.py

```diff
@@ -195,7 +195,7 @@ def distributed_main_mock(i, main, cfg, kwargs, events):
     # need to patch this seperately here, otherwise spawns won't be patched
     with patch("logging.Logger._log", partial(log_to_events, events=events)):
         with patch(
-            "metaseq.cli.train.subprocess.run",
+            "metaseq.cli.train._run_azcopy",
```

Comment:
Instead of patching the overall subprocess.run function, only mock the specific function we want to track